server/pkg/webmanager/manager.go

```diff
@@ -79,32 +70,59 @@ func RequestFile(scAddress string, networkInfo *msConfig.NetworkInfos, resourceN
 func fetchAndCache(networkInfo *msConfig.NetworkInfos, scAddress string, cache *cache.Cache, resourceName string) ([]byte, error) {
 	websiteBytes, err := website.Fetch(networkInfo, scAddress, resourceName)
 	if err != nil {
-		if website.IsNotFoundError(err, resourceName) {
-			return nil, fmt.Errorf("website %s not found: %w", scAddress, err)
-		}
-
 		return nil, fmt.Errorf("failed to fetch %s from %s: %w", resourceName, scAddress, err)
 	}
 
 	logger.Debugf("%s: %s successfully fetched with size: %d bytes", scAddress, resourceName, len(websiteBytes))
 
-	if err := cache.Save(scAddress, resourceName, websiteBytes); err != nil {
-		return nil, fmt.Errorf("failed to save %s to %s cache: %w", resourceName, scAddress, err)
-	}
+	if cache != nil {
+		err := cache.Save(scAddress, resourceName, websiteBytes)
+		if err != nil {
+			return nil, fmt.Errorf("failed to save %s to %s cache: %w", resourceName, scAddress, err)
+		}
 
-	logger.Infof("%s: %s successfully written to cache", scAddress, resourceName)
+		logger.Infof("%s: %s successfully written to cache", scAddress, resourceName)
+	}
 
 	return websiteBytes, nil
 }
 
 // Compares the last updated timestamp to the file's last modified timestamp.
+// Returns true if the file is outdated or if an error occurred. Otherwise, returns false.
 //
 //nolint:unused
-func isFileOutdated(cache *cache.Cache, scAddress string, fileName string, lastUpdated time.Time) (bool, error) {
+func isFileOutdated(cache *cache.Cache, networkInfo *msConfig.NetworkInfos, scAddress string, fileName string) (bool, error) {
+	lastUpdated, err := website.GetLastUpdateTimestamp(networkInfo, scAddress)
+	if err != nil {
+		return true, fmt.Errorf("failed to get last update timestamp: %w", err)
+	}
+
 	lastModified, err := cache.GetLastModified(scAddress, fileName)
 	if err != nil {
-		return false, fmt.Errorf("failed to get file info: %w", err)
+		return true, fmt.Errorf("failed to get file info: %w", err)
+	}
+
+	return lastModified.Before(*lastUpdated), nil
+}
+
+func ResourceExistsOnChain(network *msConfig.NetworkInfos, websiteAddress, filePath string) (bool, error) {
+	logger.Debugf("Checking if file %s exists on chain for website %s", filePath, websiteAddress)
+
+	isPresent, err := website.FilePathExists(network, websiteAddress, filePath)
+	if err != nil {
+		return false, fmt.Errorf("checking if file is present on chain: %w", err)
+	}
+
+	return isPresent, nil
+}
+
+func ResourceExistsInCache(scAddress string, resourceName string) (bool, error) {
+	logger.Debugf("Checking if resource %s exists in cache for website %s", resourceName, scAddress)
+
+	cache, err := cache.NewCache(cacheDir)
```

Comment:
it seems weird to create a new cache entity everytime you need it. The cache should be initialized when the app starts. You can look at singleton design pattern

Reply:
Will be done in https://github.com/massalabs/DeWeb/issues/147